### ISSUE TITLE
FormFieldDirectivesSpec.The 'formFieldMap' directive should not show bad performance characteristics when field names' hashCodes collide

### ISSUE THREAD

**Issue by johanandren** [akka/akka-http#2386]
#2313 reoccurring

https://jenkins.akka.io:8498/job/pr-validator-akka-http/3157/testReport/junit/akka.http.scaladsl.server.directives/FormFieldDirectivesSpec/The__formFieldMultiMap__directive_should_not_show_bad_performance_characteristics_when_field_names__hashCodes_collide/
```
org.scalatest.exceptions.TestFailedException: 4.351151441489122 was not less than 3.0
	at org.scalatest.MatchersHelper$.indicateFailure(MatchersHelper.scala:346)
	at org.scalatest.Matchers$ShouldMethodHelper$.shouldMatcher(Matchers.scala:6668)
	at org.scalatest.Matchers$AnyShouldWrapper.should(Matchers.scala:6704)
	at akka.http.scaladsl.server.directives.FormFieldDirectivesSpec.$anonfun$new$89(FormFieldDirectivesSpec.scala:281)
```

**Comment by raboof** [comment]
Maybe we should just retire those tests?

**Comment by johanandren** [comment]
Yeah, not likely that we will accidentally cause a regression there?

**Comment by jrudolph** [comment]
Another one:

https://jenkins.akka.io:8498/job/pr-validator-akka-http/3191/testReport/junit/akka.http.scaladsl.server.directives/FormFieldDirectivesSpec/The__formFieldMap__directive_should_not_show_bad_performance_characteristics_when_field_names__hashCodes_collide/

**Comment by jrudolph** [comment]
Fixed in #2424.
